documentation/src/main/wiki/Knot.md

```diff
@@ -1,5 +1,99 @@
+#Knot
 Knot is a module which defines custom step during [[request routing|KnotRouting]]. It can process custom
 fragments, invoke Adapters and redirect site visitors to a new site or an error page.
 
+##How does it work?
+Knot module gets Knot Context, does his job and responds with Knot Context. This is a very simple but 
+powerful contract which makes Knot easy to integrate and develop.
+
+Knot registers to Event Bus with an unique address and listens for Knot Context events. Each Knot is deployed
+as a separate verticle and is triggered during [[request routing|KnotRouting]].
+
+To understand what Knots really do we need to know what Knot Context is.
```

Comment:
`[Knot Context](#knot-context)` should probably be a link to `### Knot Context` anchor.